common/cards/default/single-use/chorus-fruit.ts

```diff
@@ -29,7 +26,15 @@ class ChorusFruitSingleUseCard extends SingleUseCard {
 				[`attack `, 'plain'],
 			])
 
-			player.hooks.afterAttack.remove(instance)
+			player.hooks.afterAttack.add(instance, (attack) => {
+				// Remove change active hermit from the blocked actions so it can be done once more
+				game.removeBlockedActions('game', 'CHANGE_ACTIVE_HERMIT')
```

Comment:
This needs to straight away remove the afterAttack hook so it can't get called again

Reply:
I believe nothing in the game could trigger afterAttack for a second time when you're using Chorus Fruit. Also, if I just delete it in afterAttack, a damage effect would cancel Chorus Fruit.

Reply:
The afterAttack hook is called for every section of an attack, it's bad practice for e.g. "game.removeBlockedActions('game', 'CHANGE_ACTIVE_HERMIT')" isn't run more than once for a *single* attack loop, as it wouldn't achieve anything. Therefore the afterAttack hook should still self remove.

I agree with the change made here to allow it to work for multiple attack loops, but you should also add a check to see if you've already applied the card before applying it again.

Reply:
The card is applied in onAttack, which gets deleted in onDetach when Gem deletes the card. Here, onAttack summons afterAttack which keeps the option to switch because attacks remove that option. If afterAttack removes itself, there will be nothing to reallow the option to switch, though we actually can still call it intentional because there had been no opinions about this unless I missed something in the javascript era.

Reply:
I made a PR that fixes the issues @Sense101 is referring to here [https://github.com/Rvtar/hc-tcg/pull/2](url). Using chorus fruit with Anvil, Anvil Drop, or Fire/Poison causes attack hooks to run too often.

Reply:
Thanks, I was completely tunnel-visioned on Gem+Chorus Fruit.